src/topology/inseparable.lean

```diff
@@ -184,6 +184,225 @@ lemma continuous.specialization_monotone (hf : continuous f) :
   @monotone _ _ (specialization_preorder X) (specialization_preorder Y) f :=
 λ x y h, h.map hf
 
+/-- A subset `S` of a topological space is stable under specialization
+if `x ∈ S → y ∈ S` for all `x ⤳ y`. -/
+def stable_under_specialisation (s : set X) : Prop :=
+∀ ⦃x y⦄, x ⤳ y → x ∈ s → y ∈ s
+
+/-- A subset `S` of a topological space is stable under specialization
+if `x ∈ S → y ∈ S` for all `y ⤳ x`. -/
+def stable_under_generalisation (s : set X) : Prop :=
+∀ ⦃x y⦄, y ⤳ x → x ∈ s → y ∈ s
```

Comment:
I think it's a good opportunity to develop some API for downward-closedness (and upward?) under a relation as I [suggested](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/downward-closedness.20under.20a.20general.20relation/near/285012419); you could formulate `stable_under_specialization/generalization` as downward-closedness under the specializing relation and its [swap/flip](https://leanprover-community.github.io/mathlib_docs/init/function.html#function.swap) so you could apply the same API to two use cases, and many lemmas you prove can be generalized to arbitrary relations (some only to preorders).

Reply:
I'll look into this.

Reply:
I just realized that if [specialization_preorder](https://leanprover-community.github.io/mathlib_docs/topology/inseparable.html#specialization_preorder) is an instance then you can just use [is_lower_set](https://leanprover-community.github.io/mathlib_docs/order/upper_lower.html#is_lower_set)/is_upper_set. If you're able to make them work with local attribute instance or something, then there is no need to deal with a general relation as I suggested in the Zulip topic

Reply:
I've incorporated your suggestions to shorten lemmas. I did not use them in the definitions in case the appearance of local instances causes trouble later on.